azurelinuxagent/ga/remoteaccess.py

```diff
@@ -48,108 +47,107 @@ def get_remote_access_handler(protocol):
                       is_success=False,
                       message=msg)
 
-    def handle_remote_access(self):
-        # Get JIT user accounts.
-        all_users = self.os_util.get_users()
-        existing_jit_users = set(u[0] for u in all_users if self.validate_jit_user(u[4]))
-        self.err_message = ""
-        if self.remote_access is not None:
-            goal_state_users = set(u.name for u in self.remote_access.user_list.users)
-            for acc in self.remote_access.user_list.users:
+    def _get_existing_jit_users(self):
+        all_users = self._os_util.get_users()
+        return set(u[0] for u in all_users if self._is_jit_user(u[4]))
+
+    def _handle_remote_access(self):
+        if self._remote_access is not None:
+            logger.info("Processing remote access users in goal state.")
+
+            self._check_existing_jit_users = True
+
+            existing_jit_users = self._get_existing_jit_users()
+            goal_state_users = set(u.name for u in self._remote_access.user_list.users)
+
+            for acc in self._remote_access.user_list.users:
                 try:
                     raw_expiration = acc.expiration
                     account_expiration = datetime.strptime(raw_expiration, REMOTE_USR_EXPIRATION_FORMAT)
                     now = datetime.utcnow()
                     if acc.name not in existing_jit_users and now < account_expiration:
-                        self.add_user(acc.name, acc.encrypted_password, account_expiration)
+                        self._add_user(acc.name, acc.encrypted_password, account_expiration)
                     elif acc.name in existing_jit_users and now > account_expiration:
                         # user account expired, delete it.
-                        logger.info("user {0} expired from remote_access".format(acc.name))
-                        self.remove_user(acc.name)
-                except RemoteAccessError as rae:
-                    self.err_message = self.err_message + "Error processing user {0}. Exception: {1}"\
-                        .format(acc.name, ustr(rae))
+                        logger.info("Remote access user '{0}' expired.", acc.name)
+                        self._remove_user(acc.name)
+                except Exception as e:
+                    logger.error("Error processing remote access user '{0}' - {1}", acc.name, ustr(e))
+
             for user in existing_jit_users:
                 try:
                     if user not in goal_state_users:
                         # user explicitly removed
-                        logger.info("User {0} removed from remote_access".format(user))
-                        self.remove_user(user)
-                except RemoteAccessError as rae:
-                    self.err_message = self.err_message + "Error removing user {0}. Exception: {1}"\
-                        .format(user, ustr(rae))
+                        self._remove_user(user)
```

Comment:
Don't we want to keep the log when we are removing the user?

Reply:
You are logging it when removing below on L122.

Reply:
_remove_user does the logging; 122 is a left-over, will remove it